### ISSUE TITLE
Buffer allocation using double pointer

### ISSUE THREAD

**Issue by yvn96** [halide/Halide#4474]
Hi, 
I am trying to allocate halide buffers using the host buffer pointers.
`
Void func_call(float **pin) {
    Buffer<float>input(pin, width, height, channel);
}`
This gives me error message saying no conversion from float** to float *
If I change the code and do
`Void func_call(float **pin) {
    Buffer<float>input(&pin[0][0], width, height, channel);
}`

This gets compiled fine but doesn't give me correct output of the halide optimized code. I get zeroes in some part of the output image.
Whereas if I do like the following it works fine
`Void func_call(float **pin) {
    Buffer<float>input(width, height, channel);
    for(int i=0; i<height;i++){
        for(int j=0; j<width;j++){
            int idx=i*width + j;
            input(j,i,0) = pin[0][idx];
            input(j,i,1) = pin[1][idx];
         }
    }
}`
I want to avoid this one by one copy and use the host allocated pointer. Any suggestions @abadams ?

**Comment by abadams** [comment]
So each channel of pin is potentially stored somewhere different? Halide can't really cope with arrays of pointers like that. It requires there to be an affine mapping from multi-dimensional indices down to linear memory (i.e. base_address + x * stride_x + y * stride_y + z * stride_z + ... ). 

If 'pin' takes tons of memory and you don't want to make a full copy, then you can lazily copy values into linear memory on demand using an extern producer, but it's complex: https://github.com/halide/Halide/blob/master/test/correctness/extern_producer.cpp

If there are only a small finite numbers of planes of pin, e.g. three, then you can just treat them as three separate input images. That might be your best solution.

**Comment by yvn96** [comment]
Yes, apparently each channel is stored at different locations. Copying values lazily into linear memory doesn't mean creating a new halide buffer and copying values in it one by one?
If not could you tell some resources to understand the "extern_producer.cpp" better. All I want is to stop the wastage of extra memry and time lapsed in allocating it.
Yes true, I can treat them as separate buffers also, thanks.

**Comment by abadams** [comment]
> [!NOTE]
> **Automated triage:** this issue was flagged by Claude as likely safe to close, for the
> reason below. The reasoning was checked against the current source, and a second pass
> tried and failed to refute it, but no human verified this specific text.
> **If the reason is wrong, please reopen the issue and say so** — some fraction of these
> are expected to be wrong.

**Reason (CLOSE_WONTFIX):** A support question: can a Halide Buffer wrap a float** where each channel lives at an unrelated address? It cannot, by design — halide_buffer_t has a single host pointer plus per-dimension strides, so indices must map affinely to one linear allocation. The thread was answered with the two supported options (separate buffers per plane, or an extern producer) and the reporter acknowledged them.

**Evidence:** halide_buffer_t in src/runtime/HalideRuntime.h carries a single `uint8_t *host` plus halide_dimension_t stride/min/extent per dimension; Halide::Runtime::Buffer constructors all take one pointer. test/correctness/extern_producer.cpp is the documented lazy-copy escape hatch.

Comment and close: "Halide buffers require an affine mapping from indices to a single linear allocation (halide_buffer_t has one host pointer plus strides), so an array-of-plane-pointers can't be wrapped directly. Use one Buffer per plane, or a define_extern producer that copies on demand (see test/correctness/extern_producer.cpp). Closing as working-as-designed — happy to continue the discussion in a Discussion thread."
